File: src/main/java/less/green/openpudo/rest/dto/pack/PackageSummary.java
package less.green.openpudo.rest.dto.pack;

import com.fasterxml.jackson.annotation.JsonFormat;
import less.green.openpudo.business.model.usertype.PackageStatus;
import lombok.Data;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

import java.util.Date;
import java.util.UUID;

@Data
public class PackageSummary {

    @Schema(readOnly = true)
    private Long packageId;

    @Schema(readOnly = true)
    @JsonFormat(shape = JsonFormat.Shape.STRING)
    private Date createTms;

    @Schema(readOnly = true)
    private PackageStatus packageStatus;

    @Schema(readOnly = true)
    private UUID packagePicId;

    @Schema(readOnly = true)
    private String packageName;

    @Schema(readOnly = true)
    private Long pudoId;

    @Schema(readOnly = true)
    private String businessName;

    @Schema(readOnly = true)
    private String label;

    @Schema(readOnly = true)
    private Long userId;

    @Schema(readOnly = true)
    private String firstName;

    @Schema(readOnly = true)
    private String lastName;

    @Schema(readOnly = true)
    private String customerSuffix;

}

File: lib/models/package_summary.dart
/*
 OpenPUDO - PUDO and Micro-delivery software for Last Mile Collaboration
 Copyright (C) 2020-2022 LESS SRL - https://less.green

 This file is part of OpenPUDO software.

 OpenPUDO is free software: you can redistribute it and/or modify
 it under the terms of the GNU Affero General Public License version 3
 as published by the Copyright Owner.

 OpenPUDO is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU Affero General Public License version 3 for more details.

 You should have received a copy of the GNU Affero General Public License
 version 3 published by the Copyright Owner along with OpenPUDO.
 If not, see <https://github.com/lessgreen/OpenPUDO>.
*/

import 'package:json_annotation/json_annotation.dart';

part 'package_summary.g.dart';

@JsonSerializable()
class PackageSummary {
  final int packageId;
  final DateTime? createTms;
  final String? packagePicId;
  final String? packageName;
  final PackageStatus? packageStatus;
  final int? pudoId;
  final String? businessName;
  final String? label;
  final int? userId;
  final String? firstName;
  final String? lastName;
  final String? customerSuffix;

  PackageSummary(
      {required this.packageId,
      this.createTms,
      this.packagePicId,
      this.packageName,
      this.packageStatus,
      this.pudoId,
      this.businessName,
      this.label,
      this.userId,
      this.firstName,
      this.lastName,
      this.customerSuffix});

  factory PackageSummary.fromJson(Map<String, dynamic> json) => _$PackageSummaryFromJson(json);

  Map<String, dynamic> toJson() => _$PackageSummaryToJson(this);
}

enum PackageStatus {
  @JsonValue("delivered")
  delivered,
  @JsonValue("notify_sent")
  notifySent,
  @JsonValue("notified")
  notified,
  @JsonValue("collected")
  collected,
  @JsonValue("accepted")
  accepted,
  @JsonValue("expired")
  expired,
}
